### ISSUE TITLE
kubeadm join phase preflight: [ERROR HTTPProxy]: parse https://xxxx::1:2a: invalid port ":2a" after host

### ISSUE THREAD

**Issue by hangxie** [kubernetes/kubernetes#82304]
**What happened**:
With kubadm 1.15.3, running `kubeadm join phase preflight` returned
```
I0904 00:29:27.135405    6199 checks.go:209] validating availability of port 2380
I0904 00:29:27.135453    6199 checks.go:254] validating the existence and emptiness of directory /var/lib/etcd
error execution phase preflight: [preflight] Some fatal errors occurred:
	[ERROR HTTPProxy]: parse https://xxxx::1:2a: invalid port ":2a" after host
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
```
We run same configuration without  problem under 1.15.0, 1.15.1, and 1.15.2.

**What you expected to happen**:
All patches within 1.15 maintain backward compatibility (preferred) or document all incompatibility issues (less preferred).

**How to reproduce it (as minimally and precisely as possible)**:
We are running IPv6 only with egress proxy.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.3", GitCommit:"2d3c76f9091b6bec110a5e63777c332469e0cba2", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:54Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
Openstack
- OS (e.g: `cat /etc/os-release`):
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
- Kernel (e.g. `uname -a`):
`Linux x1-master-2.XXX 3.10.0-957.27.2.el7.x86_64 #1 SMP Mon Jul 29 17:46:05 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`
- Install tools:
kubeadm
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by hangxie** [comment]
@kubernetes/sig-cluster-lifecycle-bugs

**Comment by neolit123** [comment]
a PR for that merged and will be available in 1.16:
https://github.com/kubernetes/kubernetes/pull/82267

your workaround for 1.15 is to add `--ignore-preflight-errors=HTTPProxy`
